### ISSUE TITLE
Tag removal does not sync to other MISP instances

### ISSUE THREAD

**Issue by RichieB2B** [MISP/MISP#2223]
When a tag is deleted and the event re-published, the tag is not removed from the event on downstream MISP instances. This is counter intuitive and can lead to strange situations such as events that have two TLP colors.

**Comment by mike1703** [comment]
This is still an issue in 2.4.81 and as this is counter intuitive I think deletions should be synced. There is the following audit log entry on the remote instance which shows me that at least the publish_timestamp is refreshed:
`user_id (2) => (1), publish_timestamp (1508323849) => (1508323854) `

Additionally deletions of attributes are also not synced to remote instances.
But for the deletion of attributes I get at least the following logline in the remote instances audit log once I remove the attribute:
`user_id (2) => (1), attribute_count (1) => (0), publish_timestamp (1508323920) => (1508324055) `

`attribute_count (x) => (y)`  x: number of elements in the remote instance, y: number of elements in the event on the source instance.

**Comment by wilson-santos-pf** [comment]
Any prediction on when this would be available?

**Comment by mike1703** [comment]
As of 2.4.92:
For documentation: The addition in my previous comment from 18 Oct 2017, that **deletion of attributes** is not synced seems to be **solved**

But the part about tags (neither event tags nor attribute tags) **not being deleted** on the remote instance is **still an issue**.

**Comment by iglocska** [comment]
Correct, we still don't have a system in place for the remote deletion of
tags. It would be a great to have feature but currently not the highest
priority (though I can see it rising with certain use-cases faster and
faster - such as the rise of ATT&CK usage).

**Comment by RoyClark992** [comment]
This is still an issue with MISP version 2.4.151, any update when this will be fixed or a workaround ?

**Comment by Hestat** [comment]
While performing some sync tests (v2.4.156/158) noticed that this is still an open issue. Had some tagging automation add bad tags quite some time ago, and was manually cleaned on our main MISP instance, but now see that even a sync to a new server some 6 months later the bad tags are still attached to the events. This will likely cause some good deal of confusion as some events have unrelated tags added, and no way to reasonably fix while keeping the key data.

**Comment by david74552** [comment]
Bump for update - how goes the soft-delete update?, we kind of need this.

**Comment by dragsu** [comment]
Possibly fixed in v2.4.162. I have not tested it.
```
https://www.misp-project.org/Changelog.txt 
v2.4.162 (2022-09-09)
[sync] Allow option to delete tags on event sync prior to soft-delete  tag implementation. [Tom King]
```

**Comment by david74552** [comment]
> Possibly fixed in v2.4.162. I have not tested it.
> 
> ```
> https://www.misp-project.org/Changelog.txt 
> v2.4.162 (2022-09-09)
> [sync] Allow option to delete tags on event sync prior to soft-delete  tag implementation. [Tom King]
> ```

Our analysts just tested .166 and it didn't seem to do the job. Was there a configuration flag we should have been looking at?

**Comment by dragsu** [comment]
The tooltip for  `Remove Missing Attribute Tags(not recommended)` option under Misc Settings in sync server config gives below,

`Remove any global tags from attributes on local instance that are not present on an updated event being received from the server. Any missing global tags will be removed, local tags are unaffected as is pushing events (working with Pull event).`

and I did the below test and had no luck with it.

- Configure a sync server in _Instance A_ to pull events from _Instance B_ (`Remove Missing Attribute Tags` is selected in _Instance A_)
- Create an event in _Instance B_ with a simple attribute
- Attach `tlp:green` as a global tag to the attribute
- Pull the event into Instance A, and verify `tlp:green` is present in the attribute (ok)
- Update the tlp tag of the attribute to `tlp:white` in _Instance B_ and republish it
- Pull the event again into _Instance A_ and verify only `tlp:white` is present (nok), both `tlp:white` and `tlp:green` is present in the attribute.

**Comment by tomking2** [comment]
Hi @dragsu, @acsc-davidg,

I believe this mechanism is still working for us as we expect it to. I'll have a test locally in the new year, and report back if this is in fact not the case. If so, I'll work on submitting a new PR to address any issues.

**Comment by tiagosequeira** [comment]
This is an important feature to implement the event curation workflows. The tagging mechanism is really important to do that in a proper way. I did some tests and the issue continues. When do you estimate this issue will be resolved?

**Comment by tiagosequeira** [comment]
Any news about this issue? Thanks a lot.

**Comment by RoyClark992** [comment]
Problem occurs in version 171, any update?

**Comment by flyingrobot715** [comment]
Are there any updates regarding this issue? Is this fixed in the new MISP version?

**Comment by tomking2** [comment]
I wouldn't expect it to be fixed in the new version. I'll spend a bit of time testing today, and try to get a PR to solve this issue

**Comment by JoePJisc** [comment]
We are running into this issue where events are being created with tags for monitoring, then when the analyst removes this tag and adds the blocking tag the remote event ends up with both tags and ambiguous behaviour in downstream tools.

I can see the "Remove Missing Tags" option in the server options, but note it is "not recommended".

Why isn't this recommended? What are the potential side effects?
